daemon/containerd/image_builder.go

```diff
@@ -493,9 +493,6 @@ func (i *ImageService) CreateImage(ctx context.Context, config []byte, parent st
 		return nil, err
 	}
 
-	newImage := dimage.NewImage(dimage.ID(createdImage.Target.Digest))
-	newImage.V1Image = imgToCreate.V1Image
```

Comment:
Somewhat curious; so in the old implementation we were copying the v1Image.

Are we setting the "wrong" parent? Note that _both_ V1Image and Image have a `Parent` field, and I think the second one is shadowing the first one (I still need to fully understand how those two relate);

https://github.com/moby/moby/blob/b63400fa4a122ea3575811656dccc5ab9dc2dcdd/image/image.go#L39-L48


https://github.com/moby/moby/blob/b63400fa4a122ea3575811656dccc5ab9dc2dcdd/image/image.go#L91-L100



The `NewImage()` function is used on _one_ other location now; does that location have the same issue (i.e., should that function be fixed), or are they deliberately used differently?

Reply:
The `//nolint:govet` comment is kinda telling that it's confusing; at least that one should have a comment after it to explain what (and WHY) we're doing this.

Reply:
It doesn't matter, both `Parent` fields are empty at this point.

But, good thing you bring this up, because that's actually another bug related to https://github.com/moby/moby/issues/46194 which is caused by the `Parent` field not being set there.

The graphdriver's `CreateImage` actually does this indirectly (like with the `computedId`) by first setting the parent in the `ImageStore` and then doing a `Get` from the `ImageStore` which sets the `Parent`:

https://github.com/moby/moby/blob/b63400fa4a122ea3575811656dccc5ab9dc2dcdd/daemon/images/image_builder.go#L249-L262

https://github.com/moby/moby/blob/b63400fa4a122ea3575811656dccc5ab9dc2dcdd/image/store.go#L208-L228

Reply:
But yeah, the whole thing is confusing and would benefit form some a refactor. I don't want to do this now though. And if we want to rip the whole graph driver code in some future, then maybe there's no point in doing it now.

Reply:
No, definitely don't do that as part of this PR. I mostly wanted to double-check if the "other" code paths were right (and was hoping you had that fresh in mind).

It's hella confusing for sure, and I've been staring at some of that code multiple times, trying to figure out the "what" and "why", and .. still don't fully know.